File: hejoo345/week_17/PGS_소수 찾기.js
function solution(numbers) {
    let answer = new Set();
    const arr = numbers.split("");

    // 순열 만들기
    function getPermutation(arr, fixed) {
        if (arr.length > 0) {
            for (let i = 0; i < arr.length; i++) {
                const newNum = fixed + arr[i];
                // console.log(newNum);
                const tmp = [...arr];
                tmp.splice(i, 1);

                if (isPrime(Number(newNum))) {
                    answer.add(Number(newNum));
                }
                getPermutation(tmp, newNum);
            }
        }
    }

    getPermutation(arr, '');
    return answer.size;
}

// 소수 판별하기
function isPrime(num) {
    if (num <= 1) return false;
    for (let i = 2; i * i <= num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}

File: OneK-2/week_26/PGS_소수 찾기.java
import java.util.*;

class Solution {
    static ArrayList<Integer> list = new ArrayList<Integer>();
    static boolean[] isVisited = new boolean[7];

    public int solution(String numbers) {
        int answer = 0;
        for (int i = 0; i < numbers.length(); i++) {
            makeNum(numbers, "", i + 1);
        }
        for (int i = 0; i < list.size(); i++) {
            if (isPrime(list.get(i))) answer++;
        }
        return answer;
    }

    public static boolean isPrime(int num) {
        if (num < 2) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        return true;
    }

    public static void makeNum(String numbers, String tmp, int depth) {
        if (tmp.length() == depth) {
            int num = Integer.parseInt(tmp);
            if (!list.contains(num)) list.add(num);
        }

        for (int i = 0; i < numbers.length(); i++) {
            if (!isVisited[i]) {
                isVisited[i] = true;
                tmp += numbers.charAt(i);
                makeNum(numbers, tmp, depth);
                isVisited[i] = false;
                tmp = tmp.substring(0, tmp.length() - 1);
            }
        }
    }
}